server/src/main/java/org/opensearch/transport/TransportService.java

```diff
@@ -767,7 +769,10 @@ public final <T extends TransportResponse> void sendRequest(
         final TransportResponseHandler<T> handler
     ) {
         try {
-            logger.info("Action: " + action);
+            logger.info("Action: " + action + ", as Subject: " + SecurityUtils.getSubject().getPrincipal());
+            if (!SecurityUtils.getSubject().isAuthenticated()) {
+                throw new RuntimeException("AHH!");
```

Comment:
Probably a narrower exception type you can use here.

Reply:
This will be removed - trying to get a sense of what actions are working without authentication